src/ini.cr

```diff
@@ -52,7 +52,7 @@ class INI
       end
     end
 
-    ini.delete_if { |_, v| v.empty? }
+    ini.delete_if { |k, v| k.empty? && v.empty? }
```

Comment:
No, it deletes empty sections name that have no keys: `{ "" => {} }`.

Reply:
It essentially removes the default section if it is empty. Don't know if that's needed. But if so, it could be simplified to this single case.

Reply:
If we look at the specs: `{ "section" => {} }` and `{ "" => {"key" => "value"} }` souldn't be deleted, only `{ "" => {} }`. This is the point of this PR

Reply:
However there is a maybe better to do this.

Reply:
If `ini[""].empty?`, then just `init.delete("")`?

Reply:
But why delete an empty section? I mean, this ini:

```ini
[]
```

must parse into:

```
{"" => {}}
```

and when you serialize that, it must create the same ini file. Otherwise you can't tell that the difference from that an empty file.

About what's the use case for an empty section, I don't know. But it's the more mathematically correct solution, in my opinion.

Reply:
@asterite INI files already don't roundtrip 100% on order, duplicated sections, and whitespace. `[]` is just a duplicated section of the default section, so I think this is fine.

Reply:
Whatever, it's just an arbitrary line of code, I can't understand why it's there in the first place.